src/BaseGraph.ts

```diff
@@ -0,0 +1,956 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import { Client, ResponseType } from '@microsoft/microsoft-graph-client';
+import * as MicrosoftGraph from '@microsoft/microsoft-graph-types';
+import * as MicrosoftGraphBeta from '@microsoft/microsoft-graph-types-beta';
+import { IProvider, prepScopes } from '.';
+import { MgtBaseComponent } from './components/baseComponent';
+import { Batch } from './utils/Batch';
+
+/**
+ * The interface for making graph calls
+ *
+ * @export
+ * @interface IGraph
+ */
+export interface IGraph {
```

Comment:
Is IGraph necessary here? Why not just use the abstract class BaseGraph?

Reply:
For purposes of separation of concerns. Although I admit it is wholly unnecessary at the moment. Shall I trim it and stick with BaseGraph?

Reply:
Yeah, I'd try to keep things simple unless necessary. I wasn't sure if I was missing anything, but it looks like IGraph can just be folded in BaseGraph